### PR TITLE
Newscan defaults: last-read seeding, login auto-join, wizard opt-in

### PR DESCRIPTION
**Pull request by robbiew**
## Summary

Improves the Message Newscan around the per-area **Auto Join** (newscan default) flag:

- **Last-read seeding** — new `MessageManager.SeedLastRead` seeds a user's JAM `.jlr` pointer 7 days back when an area joins their newscan, so new users see a manageable week of unread messages instead of the area's entire history. Never overwrites existing reading progress; quiet areas contribute zero unread.
- **Signup** — auto-joined areas now get seeded pointers, and the current areas/networks are recorded as offered (new per-user seen-sets: `seen_newscan_networks`, `seen_newscan_area_tags`).
- **Login auto-join for existing users** — pre-feature accounts are grandfathered silently (no prompt storm, newscan untouched). After that: new local Auto Join areas are silently added; a new network triggers a one-time Y/n prompt ("New network X is available - add to your newscan?") covering all its Auto Join areas. Declines are remembered; areas a user removes are never re-added.
- **Wizard opt-in** — the FTN wizard gains a toggleable "Newscan Default" Y/n field (default Yes) instead of hardcoding `AutoJoin: true`. V3Net gets a per-leaf `autoJoinAreas` setting (default yes, backward compatible) honored by all four v3net area-creation sites, asked once when subscribing to a new network in the in-BBS browser, and editable in the config-editor leaf editor.
- **Docs** — sysop docs and config-editor help text updated.

## Testing

- 16 new unit tests (SeedLastRead edge cases, join-plan semantics, wizard toggle key-driven tests, leaf-config default)
- `go test ./...` and `go test -race ./internal/menu/ ./internal/message/` pass
- `go vet` / `gofmt` clean

## Summary by CodeRabbit

* **New Features**
  * Added configurable Newscan auto-join defaults for message areas, FTN setup, and V3Net subscriptions, including a new “Newscan Default” option.
  * Added a “new network” prompt string for newly available networks.
* **Documentation**
  * Expanded “Area Properties” and auto-join behavior documentation for newscan defaults.
* **Bug Fixes**
  * Ensured FTN/V3Net area creation and area syncing honor the configured auto-join settings.
  * Improved login-time auto-join tracking and seeding of recent unread pointers.
* **Tests**
  * Added coverage for auto-join defaulting, join planning, and seed/last-read behavior.

### CODE REVIEW

File: internal/message/seed_lastread.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	}
+
+	// Messages are stored chronologically. Walk backward to find the newest
+	// message older than `since`; everything after it stays unread.
+	seed := 0
+	for i := total; i >= 1; i-- {
+		hdr, err := b.ReadMessageHeader(i)
+		if err != nil || hdr.Attribute&jam.MsgDeleted != 0 {
+			continue
+		}
```
[the remark this answers is not in the document]
Comment (robbiew) [reply internal/message/seed_lastread.go:64]: Fixed in 53e81c3 — SeedLastRead now walks forward to the first message dated within the window and no longer skips deleted messages, keeping the seed consistent with JAM's index-arithmetic unread counts. Added TestSeedLastRead_DeletedNewestOldMessage covering the deleted-newest quiet-area case (asserts 0 unread). The forward scan also fixes a related edge where a late-arriving old-dated echomail at a high message number could mask recent lower-numbered messages (TestSeedLastRead_LateArrivingOldMessage).

File: internal/menu/newscan_join_login.go [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+			}
+		}
+	}
+
+	u.SeenNewscanAreaTags = append(u.SeenNewscanAreaTags, plan.SeenTags...)
+	u.SeenNewscanNetworks = append(u.SeenNewscanNetworks, plan.SeenNets...)
+	if err := c.userManager.UpdateUser(u); err != nil {
+		slog.Error("failed to save newscan auto-join", "node", c.nodeNumber, "handle", u.Handle, "error", err)
+	}
+	return u, nil
```
[the remark this answers is not in the document]
Comment (robbiew) [reply internal/menu/newscan_join_login.go:112]: Fixed in 966aa20 — decisions are now persisted incrementally: silent additions and residual seen-marks save before the first prompt, and each answered network prompt saves immediately. A disconnect mid-prompt loses nothing already answered, and the unanswered network is left unmarked so it re-offers next login.

File: internal/menu/newscan_join.go [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+	NetworkNames map[string]string   // lowercase network -> display name for the prompt
+	SeenTags     []string            // tags to mark seen regardless of prompt outcome
+	SeenNets     []string            // lowercase networks to mark seen regardless of outcome
+}
+
+// buildNewscanJoinPlan computes which AutoJoin areas are new to the user.
+// canRead filters areas by the user's read access.
+func buildNewscanJoinPlan(areas []*message.MessageArea, u *user.User, canRead func(*message.MessageArea) bool) newscanJoinPlan {
+	if len(u.SeenNewscanAreaTags) == 0 && len(u.SeenNewscanNetworks) == 0 {
+		return newscanJoinPlan{Grandfather: true}
```
[the remark this answers is not in the document]
Comment (robbiew) [reply internal/menu/newscan_join.go]: Fixed in 966aa20 — added User.NewscanSeenInitialized (json newscan_seen_initialized) set by initNewscanSeen and persisted unconditionally, replacing the slice-length heuristic. Covered by TestBuildPlan_InitializedEmptyWorldJoinsLaterAreas (empty world at init, AutoJoin area added later → silently joined, not re-grandfathered).

File: internal/menu/v3net_areas.go [hunk 4]
```diff
@@ -323,7 +324,17 @@ func runV3NetAreas(c *cmdCtx, args string) (*user.User, string, error) {
...
+				// network has no leaf yet (its choice was already made
+				// if one exists).
+				autoJoin := true
+				if !v3netHasLeaf(e.RootConfigPath, ent.network) {
+					yes, perr := e.PromptYesNo(s, terminal, fmt.Sprintf("|15Add |14%s|15 areas to users' newscan by default?", ent.network), outputMode, nodeNumber, termWidth, termHeight, true)
+					if perr == nil {
+						autoJoin = yes
+					}
+				}
+				if err := v3netSubscribe(e.RootConfigPath, e.MessageMgr, ent.network, ent.hubURL, ent.area, autoJoin); err != nil {
```
[the remark this answers is not in the document]
Comment (robbiew) [reply internal/menu/v3net_areas.go:338]: Fixed in c9f6531 — a prompt error now aborts the subscribe (return nil, "", perr) instead of persisting the leaf with the unanswered default.

File: docs/sysop/messages/message-areas.md [hunk 5]
```diff
@@ -59,13 +59,33 @@ Message areas are defined in `configs/message_areas.json` as an array:
 - `max_msgs` — Maximum number of messages to retain (0 = no limit). Oldest messages are removed when the count is exceeded.
 - `max_msg_age` — Maximum message age in days (0 = no limit). Messages older than this are removed.
 - `sponsor` — Handle of the area sponsor/moderator (optional). See [Sponsor Menus](users/sponsor-menus.md).
+- `auto_join` — Flags this area as a newscan default. See [Auto Join Behavior](#auto-join-behavior) below.
 
 ### Area Types
 
 - **local** — Messages stay on this BBS only. No FTN processing.
 - **echomail** — Conference-style networked messages. The tosser imports/exports packets. Messages get MSGID, tearline, origin line, and SEEN-BY/PATH.
 - **netmail** — Point-to-point private FTN mail between addresses.
```
[the remark this answers is not in the document]
Comment (robbiew) [reply docs/sysop/messages/message-areas.md:70]: Added in c9f6531 — v3net documented in the Area Types list.

File: internal/configeditor/fields_areas.go [hunk 6]
```diff
@@ -136,7 +136,7 @@ func (m *Model) fieldsMsgArea() []fieldDef {
 			},
 		},
 		{
-			Label: "Auto Join", Help: "Automatically join new users to this area", Type: ftYesNo, Col: 3, Row: 12, Width: 1,
+			Label: "Auto Join", Help: "Newscan default: new users auto-join this area (existing users are offered new networks at login)", Type: ftYesNo, Col: 3, Row: 12, Width: 1,
```
[the remark this answers is not in the document]
Comment (robbiew) [reply internal/configeditor/fields_areas.go]: Reworded in c9f6531: "Newscan default: new users auto-join this area; local areas are added silently, new networks are offered at login".

File: internal/menu/newscan_join_login.go [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+	// Phase 1: silent local adds and areas whose network was already seen.
+	// Persist before any prompting so a disconnect at the network prompt
+	// doesn't lose decisions that required no prompt at all.
+	for _, tag := range plan.SilentTags {
+		addTag(tag)
+		u.SeenNewscanAreaTags = append(u.SeenNewscanAreaTags, tag)
+	}
+	for _, tag := range plan.ResidualSeenTags {
+		u.SeenNewscanAreaTags = append(u.SeenNewscanAreaTags, tag)
+	}
```
[the remark this answers is not in the document]
Comment (robbiew) [reply internal/menu/newscan_join_login.go]: Already fixed in 6c5107c (variadic append) — the lint check is green on the current head.
